api/src/main/java/jakarta/enterprise/inject/build/compatible/spi/SyntheticBeanBuilder.java

```diff
@@ -273,6 +273,28 @@
      */
     SyntheticBeanBuilder<T> withParam(String key, String[] value);
 
+    /**
+     * Adds an enum-valued parameter to the map of construction/destruction parameters.
+     * The parameter map is passed to the {@linkplain SyntheticBeanCreator creation} and
+     * {@linkplain SyntheticBeanDisposer destruction} function when the bean is instantiated/disposed.
+     *
+     * @param key the parameter key, must not be {@code null}
+     * @param value the parameter value
+     * @return this {@code SyntheticBeanBuilder}
+     */
+    SyntheticBeanBuilder<T> withParam(String key, Enum<?> value);
```

Comment:
Enums are pretty common, so I thought we should have a top-level way of passing enum-typed parameters. I don't think we need to have all primitive types on the top level (we have `boolean`, `int`, `long` and `double` and I think that's enough). Passing annotations is always possible, and that allows expressing everything [we want to allow].